Moblin/Various/ChatPost.swift

```diff
@@ -36,66 +36,102 @@ struct ChatHighlight {
     let kind: ChatHighlightKind
     let barColor: Color
     let image: String
-    let title: String
+    let titleSegments: [ChatPostSegment]
+
+    var title: String {
+        titleSegments.compactMap { segment in
+            segment.text ?? (segment.url != nil ? "[emote]" : nil)
+        }.joined()
+    }
 
     static func makeReply(user: String, segments: [ChatPostSegment]) -> ChatHighlight {
-        var title = String(localized: "Replying to \(user):")
+        let prefixText = String(localized: "Replying to \(user): ")
+        var replySegments: [ChatPostSegment] = []
+        var id = 0
+
+        // Add prefix text segment
+        replySegments.append(ChatPostSegment(id: id, text: prefixText))
+        id += 1
+
+        // Add original message segments (limited for preview)
+        var totalLength = prefixText.count
         for segment in segments {
             if let text = segment.text {
-                title += " \(text.trim())"
-            }
-            if title.count > 65 {
-                title += "..."
-                break
+                if totalLength + text.count > 65 {
+                    let remainingLength = 65 - totalLength
+                    if remainingLength > 3 {
+                        let truncatedText = String(text.prefix(remainingLength - 3)) + "..."
+                        replySegments.append(ChatPostSegment(id: id, text: truncatedText))
+                    } else {
+                        replySegments.append(ChatPostSegment(id: id, text: "..."))
+                    }
+                    break
+                }
+                totalLength += text.count
+                replySegments.append(ChatPostSegment(id: id, text: segment.text, url: segment.url))
+            } else if segment.url != nil {
+                if totalLength + 7 > 65 { // "[emote]" length
```

Comment:
Is `[emote]` ever part of a segment here? It's only added in the title string of what I can find.